storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -1058,8 +1077,7 @@ bool xbcloud_delete(Object_store *store, const std::string &container,
     if (!chunk_name_to_file_name(obj, file_name, idx)) {
       continue;
     }
-    if (!file_list.empty() &&
-        file_list.count(file_name.substr(backup_name.length() + 1)) < 1) {
+    if (skip_file(file_name, backup_name)) {
```

Comment:
This looks like refactor? :thinking: where exactly is the fix?

Reply:
This is indeed a refactor. Since the same logic is used on xbcloud_delete and xbcloud_download I created a function with this logic.

> where exactly is the fix?

It at xbcloud_download by calling this function